### PR TITLE
DOMA-2653 / DOMA-2654 Add IsApproved flag to Property

### PR DESCRIPTION
**Pull request by toplenboren**

### CODE REVIEW

**Review by sitozzz — APPROVED** [review]

File: apps/condo/domains/property/schema/Property.js [hunk 1]
```diff
@@ -261,6 +262,15 @@ const Property = new GQLListSchema('Property', {
             },
         },
 
+        isApproved: {
+            schemaDoc: 'Whether or not this organization can manage this property. Usually set by support. Defaults to False. Field is dropped to false if address is updated',
+            type: Checkbox,
```
Comment (pahaz) [inline apps/condo/domains/property/schema/Property.js:267]: defaultValue: false

File: apps/condo/domains/property/access/Property.js [hunk 2]
```diff
@@ -58,11 +58,23 @@ async function canManageProperties ({ authentication: { item: user }, originalIn
...
 
+async function canManageIsApprovedField ({ authentication: { item: user }, originalInput, operation, itemId }) {
+    if (user.isAdmin || user.isSupport) return true
+
+    // If user is not support, then he only can drop isApproved field
+    if ('isApproved' in originalInput) {
+        return get(originalInput, 'isApproved') === false
+    }
+
+    return true
```
Comment (pahaz) [inline apps/condo/domains/property/access/Property.js:69]:
```suggestion
    if (!get(originalInput, 'isApproved')) {
        return true
    }

    return false
```

File: apps/condo/domains/resident/schema/RegisterResidentService.js [hunk 3]
```diff
@@ -48,7 +48,7 @@ const RegisterResidentService = new GQLCustomSchema('RegisterResidentService', {
                 const [property] = await PropertyAPI.getAll(
                     context,
                     { address_i: propertyAddress, deletedAt: null },
-                    { sortBy: ['createdAt_ASC'], first: 1 }
+                    { sortBy: ['isApproved_ASC', 'createdAt_ASC'], first: 1 }
```
Comment (pahaz) [inline apps/condo/domains/resident/schema/RegisterResidentService.js:51]: we need to test it

**Review by pahaz — COMMENTED** [review]
You also need to change the logic inside `manageResidentToPropertyAndOrganizationConnections`

File: apps/condo/domains/resident/schema/RegisterResidentService.js (hunk 3 above)
Comment (pahaz) [reply apps/condo/domains/resident/schema/RegisterResidentService.js:51]: Probably, we need to use the manageResidentToPropertyAndOrganizationConnections here?

**Review by dkoviazin — APPROVED** [review]

**Comment by pahaz** [comment]
yarn workspace @app/condo makemigrations --check

File: apps/condo/domains/property/access/Property.js (hunk 2 above)
Comment (toplenboren) [reply apps/condo/domains/property/access/Property.js:69]: resolved

File: apps/condo/domains/property/schema/Property.js (hunk 1 above)
Comment (toplenboren) [reply apps/condo/domains/property/schema/Property.js:267]: resolved

File: apps/condo/domains/property/schema/Property.js [hunk 4]
```diff
@@ -261,6 +262,15 @@ const Property = new GQLListSchema('Property', {
...
+            access: {
+                create: access.canManageIsApprovedField,
```
[the remark this answers is not in the document]
Comment (toplenboren) [reply apps/condo/domains/property/schema/Property.js:269]: Resolved

File: apps/condo/domains/property/schema/Property.test.js [hunk 5]
```diff
@@ -11,6 +11,8 @@ const { makeClientWithRegisteredOrganization } = require('@condo/domains/organiz
 const { createTestTicket, updateTestTicket, ticketStatusByType } = require('@condo/domains/ticket/utils/testSchema')
 const { buildingMapJson } = require('@condo/domains/property/constants/property')
 const faker = require('faker')
+const { makeClientWithSupportUser } = require(
+    '@condo/domains/user/utils/testSchema')
```
[the remark this answers is not in the document]
Comment (toplenboren) [reply apps/condo/domains/property/schema/Property.test.js:15]: Resolved

**Comment by toplenboren** [comment]
> You also need to change the logic inside `manageResidentToPropertyAndOrganizationConnections`

It is already changed

File: apps/condo/domains/resident/schema/RegisterResidentService.js (hunk 3 above)
Comment (toplenboren) [reply apps/condo/domains/resident/schema/RegisterResidentService.js:52]: Yes, there are a test for this:

`connects property with matched address to resident, prefers isApproved == true`


```(js)
 it('connects property with matched address to resident, prefers isApproved == true', async () => {
        const adminClient = await makeLoggedInAdminClient()

        const [organization] = await registerNewOrganization(adminClient)
        const [OlderButApprovedProperty] = await createTestProperty(adminClient, organization, {
            map: buildingMapJson,
            isApproved: true,
        })

        const [organization2] = await registerNewOrganization(adminClient)
        await createTestProperty(adminClient, organization2, {
            map: buildingMapJson,
            address: OlderButApprovedProperty.address,
            addressMeta: OlderButApprovedProperty.addressMeta,
        })

        const payload = {
            address: OlderButApprovedProperty.address,
            addressMeta: OlderButApprovedProperty.addressMeta,
        }

        const [obj, attrs] = await registerResidentByTestClient(adminClient, payload)
        expect(obj.id).toMatch(UUID_RE)
        expect(obj.dv).toEqual(1)
        expect(obj.sender).toEqual(attrs.sender)
        expect(obj.v).toEqual(1)
        expect(obj.address).toEqual(attrs.address)
        expect(obj.addressMeta).toStrictEqual(attrs.addressMeta)
        expect(obj.user.id).toEqual(adminClient.user.id)
        expect(obj.property.id).toEqual(OlderButApprovedProperty.id)
        expect(obj.organization.id).toEqual(organization.id)
        expect(obj.unitType).toEqual(FLAT_UNIT_TYPE)
    })
```

**Pull request merged**